packages/core/src/scp-cf/destination.ts

```diff
@@ -53,14 +51,20 @@ export function parseDestination(destinationJson: DestinationJson | DestinationC
   return sanitizeDestination(destination);
 }
 
+function getDestinationConfig(destinationJson: DestinationJson | DestinationConfiguration): DestinationConfiguration {
+  return isDestinationJson(destinationJson) ? destinationJson.destinationConfiguration : destinationJson;
+}
+
 function validateDestinationConfig(destinationConfig: DestinationConfiguration): void {
-  if (typeof destinationConfig.URL === 'undefined') {
+  const isHttpDestination = destinationConfig.Type === 'HTTP' || typeof destinationConfig.Type === 'undefined';
```

Comment:
Minor: consider using a method isHttpDestination() with a flexible interface handling destinationConfig and destinationInput.